src/Umbraco.Web/PropertyEditors/ValueConverters/MemberPickerValueConverter.cs

```diff
@@ -27,44 +30,77 @@ public override PropertyCacheLevel GetPropertyCacheLevel(IPublishedPropertyType
             => PropertyCacheLevel.Snapshot;
 
         public override Type GetPropertyValueType(IPublishedPropertyType propertyType)
-            => typeof (IPublishedContent);
+        {
+            var isMultiple = IsMultipleDataType(propertyType.DataType);
+            return isMultiple
+                    ? typeof(IEnumerable<IPublishedContent>)
+                    : typeof(IPublishedContent);
+        }
+
+        private bool IsMultipleDataType(PublishedDataType dataType)
+        {
+            var config = ConfigurationEditor.ConfigurationAs<MemberPickerConfiguration>(dataType.Configuration);
+            return config.Multiple;
+        }
 
         public override object ConvertSourceToIntermediate(IPublishedElement owner, IPublishedPropertyType propertyType, object source, bool preview)
         {
-            var attemptConvertInt = source.TryConvertTo<int>();
-            if (attemptConvertInt.Success)
-                return attemptConvertInt.Result;
-            var attemptConvertUdi = source.TryConvertTo<Udi>();
-            if (attemptConvertUdi.Success)
-                return attemptConvertUdi.Result;
-            return null;
+            var nodeIds = source.ToString()
+                .Split(new[] { "," }, StringSplitOptions.RemoveEmptyEntries)
+                .Select(Udi.Parse)
+                .ToArray();
+
+            return nodeIds;
         }
 
         public override object ConvertIntermediateToObject(IPublishedElement owner, IPublishedPropertyType propertyType, PropertyCacheLevel cacheLevel, object source, bool preview)
         {
-            if (source == null)
-                return null;
+            var isMultiple = IsMultipleDataType(propertyType.DataType);
+
+            var udis = (Udi[])source;
```

Comment:
Can we just cast `source` to `Udi[]`? Even if there's only one?

Reply:
Not sure - at the moment it is like media picker https://github.com/umbraco/Umbraco-CMS/blob/51c1234db3d747cdb609236daabb5625f8058da0/src/Umbraco.Web/PropertyEditors/ValueConverters/MediaPickerValueConverter.cs#L72 .. but maybe it could try cast to different types, e.g. start with cast to `Udi`, then `int`?
Do we have something else in core trying resolving string as `Udi` or `Int` for legacy support?

Reply:
Many places I think, something like this: https://github.com/umbraco/Umbraco-CMS/blob/v8/contrib/src/Umbraco.Web/PropertyEditors/ValueConverters/ContentPickerValueConverter.cs#L39

Reply:
Is there a migration handling this for MediaPicker, since the value converter doesn't handle `int`?
https://github.com/umbraco/Umbraco-CMS/blob/v8/contrib/src/Umbraco.Web/PropertyEditors/ValueConverters/MediaPickerValueConverter.cs#L54-L64

If developers are migrating from v7 using `Umbraco.MediaPicker2` it probably isn't an issue (since this stored an `Udi` and was renamed to just `Umbraco.MediaPicker` before initial release of Umbraco v8). But if people used the obsolete `Umbraco.MediaPicker` in v7 which stored `int` ids and migrate to v8 I guess it is an issue with Media Picker as well?

Reply:
@Shazwazza do the recall if is neccesary to handle the `int` value in the value converter for migration from old MemberPicker (not MemberPicker2) in v7 to v8. The media picker value converter doesn't seem to handle this anyway, so I guess it would be an issue så well if migrating from v7 to v8 from old MediaPicker which stored `int` ids to the newer one storing 'Udi` ids.

Reply:
@bjarnef It's just that the memberpicker doesn't have a migration, content, media and MNTP have been converted to Udi in the v8.0.0 migration - https://github.com/umbraco/Umbraco-CMS/blob/2f978e96d4fbf84d8786c3245566c8800916a74b/src/Umbraco.Core/Migrations/Upgrade/V_8_0_0/UpdatePickerIntegerValuesToUdi.cs#L21-L23

So yes, we'll need to expect int identifiers here.

Reply:
Yeah, but it seems the migration only update the startNodeId i in the configuration. Where did it update the `int` value saved in content from MediaPicker v1 to `Udi` in v2?
The existing member picker didn't have configuration prevalues, but acted a single picker, so by default I have made it work as single picker (where media picker ny default is multi picker).

Reply:
🤦‍♂️  duh.. okay, I didn't read that very closely did I? 😂 

Anyway, would be good to test this out, upgrade from v7 with an int id stored.. not sure what will happen, maybe none of that is supported, maybe not. I really don't know.

Reply:
It seems Media Picker doesn't support it when migrating from v7 to v8 going from v1 to v2 picker.. but it the v2 media picker was already used in v7, it shouldn't cause any issues since it already store `Udi`.

Member Picker is probably used, but I guess it would be a bigger issue with Media Picker since it is used more and not all projects use Member Picker.